### DISCUSSION TITLE
How do I improve the DX when all my changes makes a full refresh?

### DISCUSSION THREAD

**Discussion by italodeandra** [freshframework/fresh#442] [Q&A]
I'm not used anymore to code updates triggering a full refresh on the page. It's been too long since Fast Refresh/HMR was introduced to us and `fresh` doesn't seem to have anything similar.

Is there any plan to implement something like that or is there any tips on how to improve my developer experience when working with `fresh`?

**Comment by nikableh** [comment]
If I correctly understand what you mean by _"updates triggering a full refresh on the page"_, the idea of only refreshing the component that needs to be updated is already marked as feature here — https://github.com/denoland/fresh/issues/316

**Reply by italodeandra** [reply]
Yes, they are discussing adding HMR to Fresh islands, but like @lucacasonato mentioned there

> Fresh will never support this for generic pages, as there is no infrastructure for rendering pages on the client (pages are always server side rendered)

But a solution for this is also something to look forward to. For example, the goal of this discussion would be to discover how could we save the state at least on dev mode on Fresh or what next steps we can look forward to.

Maybe adding the missing infrastructure for at least the dev mode. But I have no idea how this would impact the core.

When you call a framework "next-gen" you expect it to use the same or at least an improved experience that you have with the "old" ones. I understand that Fresh is not SPA so making it difficult to align with the experience that SPA gives us, but if we still expect people to switch to this framework we should give them the most reasons we can. Is there any room for discuss on this subject?
